src/System.Memory/tests/Performance/Perf.Utf8Formatter.cs

```diff
@@ -144,6 +144,105 @@ private static void FormatterUInt64(ulong value)
             }
         }
 
+        [Benchmark(InnerIterationCount = InnerCount)]
+        private static void FormatterDecimal()
+        {
+            decimal value = new decimal(1.23456789E+5);
+
+            byte[] utf8ByteArray = new byte[40];
+            Span<byte> utf8ByteSpan = utf8ByteArray;
+
+            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        Utf8Formatter.TryFormat(value, utf8ByteSpan, out int bytesWritten);
+                        TestHelpers.DoNotIgnore(value, bytesWritten);
+                    }
+                }
+            }
+        }
+
+        // Reenable commented out test cases when https://github.com/xunit/xunit/issues/1822 is fixed.
+        [Benchmark(InnerIterationCount = InnerCount)]
+        [InlineData(double.NegativeInfinity)]   // Negative Infinity
+        [InlineData(double.MinValue)]           // Min Negative Normal
+        [InlineData(-3.14159265358979324)]      // Negative pi
+        [InlineData(-2.71828182845904524)]      // Negative e
+        [InlineData(-1.0)]                      // Negative One
+        // [InlineData(-2.2250738585072014E-308)]  // Max Negative Normal
+        [InlineData(-2.2250738585072009E-308)]  // Min Negative Subnormal
+        [InlineData(-double.Epsilon)]           // Max Negative Subnormal (Negative Epsilon)
+        [InlineData(-0.0)]                      // Negative Zero
+        [InlineData(double.NaN)]                // NaN
+        [InlineData(0.0)]                       // Positive Zero
+        [InlineData(double.Epsilon)]            // Min Positive Subnormal (Positive Epsilon)
+        [InlineData(2.2250738585072009E-308)]   // Max Positive Subnormal
+        // [InlineData(2.2250738585072014E-308)]   // Min Positive Normal
+        [InlineData(1.0)]                       // Positive One
+        [InlineData(2.71828182845904524)]       // Positive e
+        [InlineData(3.14159265358979324)]       // Positive pi
+        [InlineData(double.MaxValue)]           // Max Positive Normal
+        [InlineData(double.PositiveInfinity)]   // Positive Infinity
```

Comment:
Do all of these have sufficiently different code paths / specializations that they're all worth running?  I think @adamsitnik was trying to cull the number of variations for a given perf test down to some small set, but I don't know exactly what the criteria he was using was.  Maybe we need to split things inner/outer loop, or something like that.

Reply:
Yes. This is covering each of the special input kinds for floating-point numbers:
* Zero (positive and negative)
* Infinity
* NaN
* Normal
* Subnormal

For the categories, we are then testing specific numbers that are known to be handled differently (either because they produce/consume additional digits, or they fall outside the normal input range and hit the slow path).